extensions/kafka/PublishKafka.h

```diff
@@ -0,0 +1,234 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <atomic>
+#include <condition_variable>
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "KafkaConnection.h"
+#include "KafkaProcessorBase.h"
+#include "controllers/SSLContextService.h"
+#include "core/Core.h"
+#include "core/FlowFile.h"
+#include "core/ProcessSession.h"
+#include "core/PropertyDefinition.h"
+#include "core/PropertyDefinitionBuilder.h"
+#include "core/PropertyType.h"
+#include "core/RelationshipDefinition.h"
+#include "core/logging/Logger.h"
+#include "core/logging/LoggerConfiguration.h"
+#include "rdkafka.h"
+#include "utils/ArrayUtils.h"
+#include "utils/RegexUtils.h"
+
+namespace org::apache::nifi::minifi::processors {
+
+class PublishKafka final : public KafkaProcessorBase {
+ public:
+  enum class CompressionCodecEnum { none, gzip, snappy, lz4, zstd };
```

Comment:
Please update `PROCESSORS.md`, as the allowed values of the `CompressCodec` property have changed.

Reply:
👍 https://github.com/apache/nifi-minifi-cpp/pull/1885/commits/5b8a9e84ecd1a067f39e51666f47fd98f79ba10c#diff-fd2410931e7fdc4bf8b3ce23f5f7a27c7aacdf9337320626d86d806448c90b9bR2019